satellite/satellitedb/gracefulexit_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+	"storj.io/storj/satellite/gracefulexit"
+	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+)
+
+func TestProgress(t *testing.T) {
+	// test basic graceful exit progress crud
+	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
+		ctx := context.Background()
```

Comment:
`testcontext` also tries to ensure that the tests don't deadlock or run for too long.

Reply:
Updated